packages/@lwc/template-compiler/src/parser/experimental.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import * as parse5 from 'parse5-with-errors';
+
+import {
+    CompilerDiagnostic,
+    generateCompilerDiagnostic,
+    LWCErrorInfo,
+    ParserDiagnostics,
+} from '@lwc/errors';
+
+import { parseHTML, traverseHTML, treeAdapter } from './html';
+
+function filter(node: parse5.AST.Default.Element) {
```

Comment:
This (and `filterChildNodes`) is really the only new thing being introduced here. Everything else is from the `template-compiler` parser.